File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/notice/service/dto/NoticeReq.java
package cc.nobrain.dev.userserver.domain.notice.service.dto;

import cc.nobrain.dev.userserver.domain.base.dto.FileDto;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import java.util.List;

@Getter
@JsonIgnoreProperties(ignoreUnknown = true)
public class NoticeReq {

    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Search {
        private String title;
        private String createMember;
    }

    @Getter
    public static class Create {
        private String title;
        private String content;
        private Boolean sticky;
        private List<FileDto> attachments;
    }
}

File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/notice/service/dto/NoticeReq.kt
package cc.nobrain.dev.userserver.domain.notice.service.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

@JsonIgnoreProperties(ignoreUnknown = true)
data class NoticeReq(val search: Search?, val create: Create?) {

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Search(
        var title: String?,
        var createMember: String?
    )

    data class Create(
        @NotNull
        var title: String,

        @NotNull
        @Size(min = 1, max = 2000)
        var content: String,

        var sticky: Boolean? = false,
        var attachments: List<Any>?
    )
}